src/conv/invokers/mlir_impl_gemm.cpp

```diff
@@ -69,6 +69,32 @@ auto PermuteDimStride(const std::vector<size_t>& dims, const std::vector<size_t>
     return std::make_tuple(sorted_dims, sorted_strides);
 };
 
+auto InsertGToDimStride(const std::string& layout,
+                        char dim,
+                        const std::vector<size_t>& dims,
+                        const std::vector<size_t>& strides,
+                        int group_count)
+{
+    auto dims_with_g    = dims;
+    auto strides_with_g = strides;
+    std::size_t index   = layout.find(dim);
+    if(index == std::string::npos)
+        MIOPEN_THROW("Failed to find channel in the layout");
+
+    // For dimensions,
+    //    Insert an additional dimension g before channel
+    //    Amend the channel size to be channel_size / group_count
+    dims_with_g.insert(dims_with_g.begin() + index, group_count);
+    dims_with_g[index + 1] = dims[index] / group_count;
```

Comment:
[Recommendaion] You can:
```suggestion
    dims_with_g[index + 1] /= group_count;
```
And if you move this above `insert()`, then
```suggestion
    dims_with_g[index] /= group_count;
```